### ISSUE TITLE
[YOUTUBEJS] [Parser]: InnertubeError: ShortsLockupView not found!

### ISSUE THREAD

**Issue by Tarekthedeveloper** [LuanRT/YouTube.js#747]
**While playing any song the song stopped and this is the console log debug  :**
This is a bug, want to help us fix it? Follow the instructions at https://github.com/LuanRT/YouTube.js/blob/main/docs/updating-the-parser.md or report it at https://github.com/LuanRT/YouTube.js/issues!
Introspected and JIT generated this class in the meantime:
class ShortsLockupView extends YTNode {
  static type = 'ShortsLockupView';

  entity_id: string;
  accessibility_text: string;
  thumbnail: {
    sources: {
      url: string,
      width: number,
      height: number
    }[]
  };
  on_tap: {
    innertube_command: NavigationEndpoint
  };
  inline_player_data: {
    on_visible: {
      innertube_command: NavigationEndpoint
    }
  };
  menu_on_tap: {
    innertube_command: NavigationEndpoint
  };
  index_in_collection: number;
  menu_on_tap_a11y_label: string;
  overlay_metadata: {
    primary_text: {
      content: string
    },
    secondary_text: {
      content: string
    }
  };
  logging_directives: {
    tracking_params: string,
    visibility: {
      types: string
    },
    enable_displaylogger_experiment: boolean
  };

  constructor(data: RawNode) {
    super();
    this.entity_id = data.entityId;
    this.accessibility_text = data.accessibilityText;
    this.thumbnail = {
      sources: data.thumbnail.sources.map((item: any) => ({
        url: item.url,
        width: item.width,
        height: item.height
      }))
    };
    this.on_tap = {
      innertube_command: new NavigationEndpoint(data.onTap.innertubeCommand)
    };
    this.inline_player_data = {
      on_visible: {
        innertube_command: new NavigationEndpoint(data.inlinePlayerData.onVisible.innertubeCommand)
      }
    };
    this.menu_on_tap = {
      innertube_command: new NavigationEndpoint(data.menuOnTap.innertubeCommand)
    };
    this.index_in_collection = data.indexInCollection;
    this.menu_on_tap_a11y_label = data.menuOnTapA11yLabel;
    this.overlay_metadata = {
      primary_text: {
        content: data.overlayMetadata.primaryText.content
      },
      secondary_text: {
        content: data.overlayMetadata.secondaryText.content
      }
    };
    this.logging_directives = {
      tracking_params: data.loggingDirectives.trackingParams,
      visibility: {
        types: data.loggingDirectives.visibility.types
      },
      enable_displaylogger_experiment: data.loggingDirectives.enableDisplayloggerExperiment
    };
  }
}

    at ERROR_HANDLER (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:13234:30)
    at createRuntimeClass (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:12400:3)
    at generateRuntimeClass (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:12443:19)
    at parseItem (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:13549:75)
    at Object.parseArray (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:13591:22)
    at new ReelShelf (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:4077:33)
    at parseItem (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:13571:22)
    at Object.parseArray (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:13591:22)
    at new ItemSection (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:8296:36)
    at parseItem (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:13571:22) {
  date: 2024-09-11T00:13:37.887Z,
  version: '10.4.0'
}

### Failure Logs

```shell
This is a bug, want to help us fix it? Follow the instructions at https://github.com/LuanRT/YouTube.js/blob/main/docs/updating-the-parser.md or report it at https://github.com/LuanRT/YouTube.js/issues!
Introspected and JIT generated this class in the meantime:
class ShortsLockupView extends YTNode {
  static type = 'ShortsLockupView';

  entity_id: string;
  accessibility_text: string;
  thumbnail: {
    sources: {
      url: string,
      width: number,
      height: number
    }[]
  };
  on_tap: {
    innertube_command: NavigationEndpoint
  };
  inline_player_data: {
    on_visible: {
      innertube_command: NavigationEndpoint
    }
  };
  menu_on_tap: {
    innertube_command: NavigationEndpoint
  };
  index_in_collection: number;
  menu_on_tap_a11y_label: string;
  overlay_metadata: {
    primary_text: {
      content: string
    },
    secondary_text: {
      content: string
    }
  };
  logging_directives: {
    tracking_params: string,
    visibility: {
      types: string
    },
    enable_displaylogger_experiment: boolean
  };

  constructor(data: RawNode) {
    super();
    this.entity_id = data.entityId;
    this.accessibility_text = data.accessibilityText;
    this.thumbnail = {
      sources: data.thumbnail.sources.map((item: any) => ({
        url: item.url,
        width: item.width,
        height: item.height
      }))
    };
    this.on_tap = {
      innertube_command: new NavigationEndpoint(data.onTap.innertubeCommand)
    };
    this.inline_player_data = {
      on_visible: {
        innertube_command: new NavigationEndpoint(data.inlinePlayerData.onVisible.innertubeCommand)
      }
    };
    this.menu_on_tap = {
      innertube_command: new NavigationEndpoint(data.menuOnTap.innertubeCommand)
    };
    this.index_in_collection = data.indexInCollection;
    this.menu_on_tap_a11y_label = data.menuOnTapA11yLabel;
    this.overlay_metadata = {
      primary_text: {
        content: data.overlayMetadata.primaryText.content
      },
      secondary_text: {
        content: data.overlayMetadata.secondaryText.content
      }
    };
    this.logging_directives = {
      tracking_params: data.loggingDirectives.trackingParams,
      visibility: {
        types: data.loggingDirectives.visibility.types
      },
      enable_displaylogger_experiment: data.loggingDirectives.enableDisplayloggerExperiment
    };
  }
}

    at ERROR_HANDLER (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:13234:30)
    at createRuntimeClass (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:12400:3)
    at generateRuntimeClass (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:12443:19)
    at parseItem (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:13549:75)
    at Object.parseArray (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:13591:22)
    at new ReelShelf (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:4077:33)
    at parseItem (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:13571:22)
    at Object.parseArray (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:13591:22)
    at new ItemSection (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:8296:36)
    at parseItem (C:\Users\User\Desktop\Coding\MusicB\Musicplayer\node_modules\youtubei.js\bundle\node.cjs:13571:22) {
  date: 2024-09-11T00:13:37.887Z,
  version: '10.4.0'
}
```

### Expected behavior

Fix the bug !

### Current behavior

 >The YouTube.js library fails to recognize a specific component (ShortsLockupView), which leads to a breakdown in parsing YouTube Shorts-related content.
> As the class is dynamically generated based on the introspection, it results in errors when trying to parse specific elements like thumbnails, menu actions, or logging directives for YouTube Shorts.

### Version

Default

### Checklist

- [X] I am running the latest version.
- [X] I checked the documentation and found no answer.
- [X] I have searched the existing issues and made sure this is not a duplicate.
- [X] I have provided sufficient information.

**Comment by sgebr01** [comment]
This is a duplicate of issue #743 and a fix is already underway with pull request #746. A release should be coming sometime soon.

**Comment by hextor1** [comment]
> This is a duplicate of issue #743 and a fix is already underway with pull request #746. A release should be coming sometime soon.

@sgebr01  I am using LuanRT/Youtubei.js also facing this ShortslockupVIew error how to solve this and when new version will be available in repo?

**Comment by absidue** [comment]
Duplicate of #743
